python/tank/platform/engine.py

```diff
@@ -1420,6 +1425,61 @@ def _emit_log_message(self, handler, record):
         """
         # default implementation doesn't do anything.
 
+    def _ensure_core_fonts_loaded(self):
+        """
+        Loads the Shotgun approved fonts that are bundled with tk-core.
+
+        This method ensures that the Shotgun approved fonts bundled with core
+        are loaded into Qt's font database. This allows them to be used by apps
+        for a consistent look and feel.
+
+        If a QApplication exists during engine initialization, it is not
+        necessary to call this method. Similarly, subclasses that make use of
+        core's bundled dark look and feel will have the bundled fonts loaded
+        automatically.
+        """
+
+        from sgtk.platform.qt import QtGui
+
+        # if the fonts have been loaded, no need to do anything else
+        if self.__fonts_loaded:
+            return
+
+        if not QtGui.QApplication.instance():
+            # there is a QApplication, so we can load fonts.
+            return
+
+        # fonts dir in the core resources dir
+        fonts_parent_dir = self.__get_platform_resource_file("fonts")
+
+        # in the parent directly, get all the font-specific directories
+        for font_dir_name in os.listdir(fonts_parent_dir):
```

Comment:
Made this just look through the fonts directory and load whatever is there. Maybe we should explicitly load? My thought was that if we decided to switch the default SG font at some point or add another font for our apps, we would just drop them in to the resources directory. We'd need to keep the old ones around anyway, so maybe this is ok.